src/primitives.rs

```diff
@@ -3,11 +3,11 @@ use std::fmt;
 
 pub use ssz_rs::prelude::Root;
 
-pub type Slot = u64;
-pub type Epoch = u64;
-pub type CommitteeIndex = u64;
-pub type ValidatorIndex = u64;
-pub type Gwei = u64;
+pub type Slot = usize;
+pub type Epoch = usize;
```

Comment:
something to consider here is that `Slot` and `Epoch` will _definitely_ get serialized and if this code runs on a machine where the word size is not 64bits then that machine will not be able to follow the consensus....

i'd lean towards keeping these as `u64` and if at some point they need to be converted into `usize`, e.g. they get pulled into some bounds computations, then we can cast there

Reply:
fixed